src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/CustomiseHearingBundlePreparer.java

```diff
@@ -47,26 +49,54 @@ public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCa
 
         AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
 
-        prepareDocuments(asylumCase, HEARING_DOCUMENTS, CUSTOM_HEARING_DOCUMENTS);
-        prepareDocuments(asylumCase, LEGAL_REPRESENTATIVE_DOCUMENTS, CUSTOM_LEGAL_REP_DOCUMENTS);
-        prepareDocuments(asylumCase, ADDITIONAL_EVIDENCE_DOCUMENTS, CUSTOM_ADDITIONAL_EVIDENCE_DOCUMENTS);
-        prepareDocuments(asylumCase, RESPONDENT_DOCUMENTS, CUSTOM_RESPONDENT_DOCUMENTS);
+        Optional<YesOrNo> maybeCaseFlagSetAsideReheardExists = asylumCase.read(CASE_FLAG_SET_ASIDE_REHEARD_EXISTS,YesOrNo.class);
+
+        boolean isCaseReheard = maybeCaseFlagSetAsideReheardExists.isPresent()
+                && maybeCaseFlagSetAsideReheardExists.get() == YesOrNo.YES;
+
+        if (isCaseReheard) {
+            prepareCustomDocuments(asylumCase, FTPA_APPELLANT_EVIDENCE_DOCUMENTS, CUSTOM_FTPA_APPELLANT_EVIDENCE_DOCS);
+            prepareCustomDocuments(asylumCase, FTPA_RESPONDENT_EVIDENCE_DOCUMENTS, CUSTOM_FTPA_RESPONDENT_EVIDENCE_DOCS);
+            prepareCustomDocuments(asylumCase, FTPA_APPELLANT_DOCUMENTS, CUSTOM_FTPA_APPELLANT_DOCS);
+            prepareCustomDocuments(asylumCase, FTPA_RESPONDENT_DOCUMENTS, CUSTOM_FTPA_RESPONDENT_DOCS);
+            prepareCustomDocuments(asylumCase, FINAL_DECISION_AND_REASONS_DOCUMENTS, CUSTOM_FINAL_DECISION_AND_REASONS_DOCS);
+            prepareCustomDocuments(asylumCase, REHEARD_HEARING_DOCUMENTS, CUSTOM_REHEARD_HEARING_DOCS);
+            prepareCustomDocuments(asylumCase, ADDENDUM_EVIDENCE_DOCUMENTS, CUSTOM_APP_ADDENDUM_EVIDENCE_DOCS);
+            prepareCustomDocuments(asylumCase, ADDENDUM_EVIDENCE_DOCUMENTS, CUSTOM_RESP_ADDENDUM_EVIDENCE_DOCS);
+
+            //if (asylumCase.read(ADDENDUM_EVIDENCE_DOCUMENTS).isPresent()) {
```

Comment:
this can be removed

Reply:
these collections are used for the customisation of the hearing.